src/common/users/services/users.interface.ts

```diff
@@ -1,10 +1,14 @@
+import { DeepPartial } from 'typeorm';
+
 import { CreateUserDto } from '../dto/create-user.dto';
 import { User } from '../entities/user.entity';
 
 export interface IUsersService {
-  create(createUserDto: CreateUserDto);
+  create(createUserDto: CreateUserDto): Promise<User>;
   findById(id: string): Promise<User | null>;
+  findByIdOrFail(id: string): Promise<User>;
```

Comment:
`findById` should have `findByIdOrFail` behaviour by default. We should not have `orFail` methods exposed by services. If we expect to fail, we should catch the error, but it's not the case anywhere. 
Let's keep only `findById` with fail behaviour